src/test/scala/org/apache/spark/sql/BebeFunctionsSpec.scala

```diff
@@ -31,13 +56,30 @@ class BebeFunctionsSpec
     }
 
     it("returns the beginning of the day in a specific timezone") {
+      println("timezone")
+      val testTimezone         = "America/Bahia"
+      val defaultTimezone      = TimeZone.getDefault()
+      val sparkDefaultTimezone = spark.conf.get("spark.sql.session.timeZone")
+      println(defaultTimezone.getDisplayName())
+      println(sparkDefaultTimezone)
+      TimeZone.setDefault(TimeZone.getTimeZone(testTimezone))
+      spark.conf.set("spark.sql.session.timeZone", testTimezone)
       val df = Seq(
-        (Timestamp.valueOf("2020-01-15 08:01:32"), Timestamp.valueOf("2020-01-14 21:00:00")),
-        (Timestamp.valueOf("2020-01-20 23:03:22"), Timestamp.valueOf("2020-01-20 21:00:00")),
+        (
+          valueOf("2020-01-15 08:01:32", testTimezone),
+          valueOf("2020-01-14 21:00:00", testTimezone)
+        ),
+        (
+          valueOf("2020-01-20 23:03:22", testTimezone),
+          valueOf("2020-01-20 21:00:00", testTimezone)
+        ),
         (null, null)
       ).toDF("some_time", "expected")
-        .withColumn("actual", beginningOfDay(col("some_time"), Some("UTC")))
+        .withColumn("actual", beginningOfDay(col("some_time"), "UTC"))
```

Comment:
@alfonsorr - do you think we should refactor `beginningOfDay` to be invoked like `beginningOfDay(col("some_time"), col("UTC"))`.  Interested in your thoughts.  I normally think it's better to pass in column arguments to these functions.

Reply:
It looks like it cant be passed as a column the timezone parameter. 😞 At least in the spark implementation.